bin_sort/bin_sort.c

```diff
@@ -0,0 +1,63 @@
+/* Following algorithm sorts the input array in ascending order
+ * Time Complexity O(n)-> avg case and O(n^2)-> Worst Case
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+struct bucket {
+	int count;
+	int* value;
+};
+
+int comparator(const void* first, const void* second) {
```

Comment:
You are type casting a `void *` into an `int *`. Instead of that pass `int *` in the definition itself.